### PR TITLE
feat: [week-6] update and translate data structures task description

### PR DESCRIPTION
**Pull request by elukashova**
Translated the Data Structures module and add English-language materials.

### CODE REVIEW

**Comment by valerydluski** [comment]
@elukashova, could you please pull the changes from master and run prettier?

**Comment by elukashova** [comment]
@valerydluski done

File: stage1/modules/data-structures/README.md [hunk 1]
```diff
@@ -1,48 +1,58 @@
...
-
-## Дополнительные материалы
-
-1. Больше методов массива: [видео на YouTube](https://youtu.be/d8c-JgbpMHs), [документация MDN](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array)
-2. Интересные обходы массивов: [видео на YouTube](https://youtu.be/jM7aTyncf8Y)
-3. Оценка сложности алгоритма: [ссылка 1](https://tproger.ru/articles/computational-complexity-explained/), [ссылка 2](https://tproger.ru/translations/algorithms-and-data-structures/)
-4. [Big O нотация и AT](https://www.youtube.com/watch?v=luExFPwQVOY)
+# Module Topic: Algorithms and Data Structures (Part 1)
+
+## Module content:
```
Comment (valerydluski) [inline stage1/modules/data-structures/README.md:3]:
```suggestion

## Module Overview

This module serves as an introduction to the fundamental concepts of algorithms and data structures in programming. It aims to provide students with a foundational understanding of algorithm efficiency metrics, such as Big O notation, and the use of various basic data structures. The module covers arrays, matrices, stacks, queues, pointers, linked lists, and binary search trees (BSTs). Through this module, students will not only learn the theoretical aspects but also apply these concepts in solving typical problems.

## Module Content
```

File: stage1/modules/data-structures/README.md [hunk 2]
```diff
@@ -1,48 +1,58 @@
...
+You need to solve the proposed algorithmic tasks.
+Attention!
+
+- It is not expected for students to solve all of the tasks.
+- Copying code from other students (current or previous cohorts) is not allowed.
+- Think carefully before submitting someone else's code; is it worth the risk of being expelled for 3% of the total grade?
+
+### Test
+
+You need to complete the "Test Algorithms & Data structures" in RS APP > Auto Test
```
Comment (valerydluski) [inline stage1/modules/data-structures/README.md:41]:
```suggestion
You need to complete the "[St1] Test Algorithms & Data structures" in RS APP > Auto Test
```

**Review by valerydluski — APPROVED** [review]

File: stage1/modules/data-structures/README.md (hunk 1 above)
Comment (elukashova) [reply stage1/modules/data-structures/README.md]: done

File: stage1/modules/data-structures/README.md (hunk 2 above)
Comment (elukashova) [reply stage1/modules/data-structures/README.md]: done

**Review by valerydluski — APPROVED** [review]

**Pull request merged**